### ISSUE TITLE
Integration of polyfill for node's `process`

### ISSUE THREAD

**Issue by keichinger** [Becklyn/kaba#42]
Unfortunately there are quite a few libs out there that sooner or later rely on node's `process` module, which isn't available in a normal web environment.

A few from our goto vendors are among the offenders:
- Algolia: https://github.com/algolia/algoliasearch-client-javascript/search?q=process&unscoped_q=process
- Preact: https://github.com/developit/preact/search?q=process&unscoped_q=process

Obviously, in a perfect world none of our vendors would explicitly rely on internal node modules, though that's somewhat unrealistic right now.

Since we've decided to disable WebPack's default polyfills (see https://github.com/Becklyn/kaba/blob/master/lib/Kaba.js#L506) we run into runtime errors and are forced to opt-out of a few features that I'd definitely like to use (e.g. `preact-debug`).

One way to circumvent this is to use a custom plugin, like mentioned here: https://github.com/developit/preact#runtime-error-checking

I'm not too sure about you, but I want this problem gone, the only question I have is how to best tackle this problem:

- Use the custom webpack plugin with a few more properties (and some logic that determines the environment based on the kaba CLI args): https://github.com/developit/preact#runtime-error-checking 
- Integrate webpack's polyfill, in an possibly even slimmed down version, either directly here or ship it as mojave module: https://github.com/webpack/node-libs-browser/blob/master/mock/process.js

What's your opinion on this topic @apfelbox?

**Comment by keichinger** [comment]
Changing the `node` option (see https://github.com/Becklyn/kaba/blob/master/lib/Kaba.js#L506) from `false` to
```js
node: {
    process: "mock",
},
```

results in the following build:

## with node: false
![image](https://user-images.githubusercontent.com/439899/47430860-d3042600-d79a-11e8-9724-cd9ff47226bc.png)

## with node: { process: "mock" }
![image](https://user-images.githubusercontent.com/439899/47430886-e44d3280-d79a-11e8-8e9b-5378290cdf8c.png)


So effectively the only difference is that `vendors~app~mayd.45c1933ad4c9400def38.js` grows by 5 KiB.

**Comment by apfelbox** [comment]
I just looked at your examples and they don't really use `process` anywhere.

### preact

Preact uses it in the tests (that don't matter to us) and to read env variables. At least `NODE_ENV` is already set via the [DefinePlugin](https://webpack.js.org/plugins/define-plugin/) in webpack, so nothing to do here.


### algolia
Algolia as well mainly uses it in tests (don't care) and for reading env variables (that we can set via the `DefinePlugin`). There is one case of `process.version` though, where we would need to take a closer look.

So, to sum it up: not worth the 5kB.
(especially as reading env variables can only be done in build time, not run time)

**Comment by keichinger** [comment]
Algolia uses it in their AlgoliaSearchCore.js (see https://github.com/algolia/algoliasearch-client-javascript/blob/develop/src/AlgoliaSearchCore.js#L12-L14), which is also the reason why it failed to run locally without manually hacking the mock in.

![image](https://user-images.githubusercontent.com/439899/47485199-eddba680-d83d-11e8-8efd-b92d66a5f064.png)

Either way, I've filed an PR (#43) to make it configurable, so this is not an issue then :)

**Comment by apfelbox** [comment]
Yes, as I previously wrote:

> and for reading env variables (that we can set via the `DefinePlugin`)

This is exactly the case of reading env variables.
So as we have no use case here, we should not make it configurable (yet?) to not bloat our project with 5kB dead code (as reading env variables in production doesn't make sense in the browser).

**Comment by apfelbox** [comment]
Suspended for now, see previous discussion.
